docs/reference/c++_support.rst

```diff
@@ -0,0 +1,165 @@
+extensions and restrictions to consider. 
+
+``__global__`` functions
+-------------------------------------------------------------------------------
+
+Functions that work as an entry point for device execution are called kernels and are
+specified with the ``__global__`` qualifier. Calling these functions happen with the
+triple chevron operator: ``<<< >>>``. Kernel functions return type must be ``void``, they
+can't be ``constexpr``, can't have a parameter of type ``std::initializer_list`` or
+``va_list``, can't have a parameter of rvalue reference type. Kernels can have variadic
+template parameters, but only one pack, and it must be the last in the template parameter
+list. 
+
+Device Space Memory Specifiers
+-------------------------------------------------------------------------------
+
+To specify whether a variable is allocated in host or device memory, HIP has qualifiers
+called device space memory specifiers. These are ``__device__``, ``__shared__``,
+``__managed__`` and ``__constant__``. These are meant to specify what memory is allocated
+for a variable. ``__device__`` and ``__constant__`` specifier are used for static
+variables and they will be allocated on the HIP devices global memory, with the only
+difference that the ``__constant__`` variables can't be changed from the device code.
+``__shared__`` is used in device code, to allocate the variable on the shared memory,
+which is available for all threads in a block. ``__managed__`` is also used on global
+variables. A managed variable is emitted as an undefined global symbol in the device
+binary and is registered by ``__hipRegisterManagedVariable`` in init functions. The HIP
+runtime allocates managed memory and uses it to define the symbol when loading the device
+binary. A managed variable can be accessed in both device and host code. It is important
+to know where each variable is because they will be only available in certain places.
+Generally, variables allocated in the host memory will not be available from device code
+and variables allocated in the device memory will not be available from host code. This
+can be an issue mostly with pointers in host memory, which are pointing to device memory.
+Dereferencing these will cause a segmentation fault.
+
+Exception handling
+-------------------------------------------------------------------------------
+
+An important difference between the host and device code is exception handling. In device
+code it is not available, error handling needs to be done with return codes. This is
+because of the hardware architecture does not allow for this kind of control flow.
+
+Kernel parameters
+-------------------------------------------------------------------------------
+
+There are some restrictions on kernel function parameters. They cannot be passed by
+reference, as these functions run on the device, but are called from the host. Also,
+variable number of arguments is not allowed.
```

Comment:
There are some restrictions on kernel function parameters. They cannot be passed by reference, because these functions are called from the host but run on the device. Also, a variable number of arguments is not allowed.